src/tests/class_tests/openms/source/FileHandler_test.cpp

```diff
@@ -294,11 +293,14 @@ PeakMap exp;
 fh.loadExperiment(OPENMS_GET_TEST_DATA_PATH("MzMLFile_1.mzML"), exp);
 
 //test mzML
-String filename;
-NEW_TMP_FILE(filename)
-fh.storeExperiment(filename, exp);
+String filename, filename2;
+NEW_TMP_FILE_EXT(filename, ".mzML");
+fh.storeExperiment(filename, exp, {FileTypes::MZML}, ProgressLogger::NONE);
 TEST_EQUAL(fh.getTypeByContent(filename), FileTypes::MZML)
 
+//Test that we throw an exception when given a bogus type restriction
+TEST_EXCEPTION(Exception::UnableToCreateFile, fh.storeExperiment(filename, exp, {FileTypes::SIZE_OF_TYPE}))
```

Comment:
I would expect an exception here, simply because the Filter is invalid -- even before dealing with I/O.
Maybe use another bogus type.

And yes, `Exception::InvalidFileType` or sth along those lines may be more expressive?

Reply:
@timosachsenberg do we have a "best practices" for adding new Exceptions? I've just stayed away from it thus far, but this seems like a reasonable case.

Reply:
Best practice is to have an exception for distinct, conceptually different error states. Let's check how accurately an existing exception represents the error state. The closest I found is InvalidValue (which is pretty general).
More specific would be what @cbielow proposed Exception::InvalidFileType .

It's not super clear-cut, but in this particular case I would probably create the new Exception.